olp-cpp-sdk-core/include/olp/core/http/Network.h

```diff
@@ -83,5 +83,10 @@ class CORE_API Network {
   virtual void Cancel(RequestId id) = 0;
 };
 
+/**
+ * @brief Create default Network implementation.
+ */
+CORE_API std::unique_ptr<Network> CreateDefaultNetwork();
```

Comment:
I presume task description is old. This should be part of the OlpClientSettingsFactory class.